mmf/datasets/builders/okvqa/dataset.py

```diff
@@ -0,0 +1,56 @@
+from typing import Type, Union
+
+import torch
+
+from mmf.common.sample import Sample
+from mmf.common.typings import MMFDatasetConfigType
+from mmf.datasets.builders.okvqa.database import OKVQAAnnotationDatabase
+from mmf.datasets.mmf_dataset import MMFDataset
+
+
+class OKVQADataset(MMFDataset):
+    def __init__(
+        self,
+        config: MMFDatasetConfigType,
+        dataset_type: str,
+        index: int,
+        *args,
+        **kwargs,
+    ):
+        super().__init__("okvqa", config, dataset_type, index, *args, **kwargs)
+
+    def build_annotation_db(self) -> Type[OKVQAAnnotationDatabase]:
+        annotation_path = self._get_path_based_on_index(
+            self.config, "annotations", self._index
+        )
+        return OKVQAAnnotationDatabase(self.config, annotation_path)
+
+    def get_image_path(self, image_id: Union[str, int]) -> str:
+        if self.dataset_type == "train":
+            image_path = f"COCO_train2014_{str(image_id).zfill(12)}"
```

Comment:
".jpg" should be appended to the image_path both for `dataset_type` train and val